tracing-attributes/src/lib.rs

```diff
@@ -524,22 +527,46 @@ fn gen_block(
                             Err(e)
                         }
                     }
-                }, __tracing_attr_span).await
+                };
+                if tracing::level_enabled!(#level) {
+                    tracing::Instrument::instrument(
+                        fut,
+                        __tracing_attr_span
+                    )
+                    .await
+                } else {
+                    fut.await
+                }
```

Comment:
actually, we could probably _also_ do the optimization of not creating the span at all if the level is disabled:
```suggestion
                if tracing::level_enabled!(#level) {
                   let __tracing_attr_span = #span; 
                   if !span.is_disabled() {
                        return tracing::Instrument::instrument(
                            fut,
                            __tracing_attr_span
                        )
                        .await;
                }
                
                fut.await
```